microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -145,6 +145,19 @@ def set_dpdk_branch(self, dpdk_branch: str) -> None:
         self._dpdk_branch = dpdk_branch
 
     def get_dpdk_version(self) -> VersionInfo:
+        if not hasattr(self, "_dpdk_version_info"):
```

Comment:
I'm not sure when the attribute is not here. But it looks the `_install` can set it. The install progress is pretty complex, and not sure if the following install steps is the same as current logic. Anyway, if the fix works well, please resolve my comments and merge it.

Reply:
this happened when the following case reuse this tool, at that time, _install is not invoked, so _dpdk_version_info is not initialized.
Yes, this is complex when reinitialized _dpdk_version_info, any suggestions to simplify it? @mcgov

Reply:
I see, the reason is the DpdkTestpmd is not initialized with a standard way, so it's behavior is weird. (testpmd = DpdkTestpmd(node)). Now, LISA supports to initialize with parameters, please try to update the logic to use `node..tools.get()` to replace it.

Reply:
sorry, I didn't understand, I need implement initialize with parameters, then node..tools.get() is for ?

Reply:
Please search the code for "testpmd = DpdkTestpmd(node)". It's not a right way to initialize a tool, so the tool object is not hold by node. The historical reason is this tool need pass parameters when initializing. Now, the node.tools.get() can pass in parameters, and can be used to create a tool like normal way. Once it's changed like this way, the variable will be available always.

Reply:
when pass like this way `node.tools.get(DpdkTestpmd, dpdk_source=dpdk_source, dpdk_branch=dpdk_branch)`, I can see the values in https://github.com/microsoft/lisa/blob/main/lisa/executable.py#L583. after that, they got lost, I can't add extra parameters for _initialize of tool DpdkTestpmd

Reply:
It needs to receive those values from the __init__ method of the tool. Check how it's used by subclass factory.

Reply:
update